packages/dev/core/src/Materials/PBR/pbrBaseMaterial.ts

```diff
@@ -889,6 +890,11 @@ export abstract class PBRBaseMaterial extends PushMaterial {
      */
     public readonly detailMap: DetailMapConfiguration;
 
+    /**
+     * Defines the decal map parameters for the material.
+     */
+    public readonly decalMap: DecalMapConfiguration;

```

Comment:
I am not sure it should if it should be added by default ???

Reply:
We could do it, but it would be harder for people to use the functionality as they would have to explicitely create the plugin for all materials they want to support.

Also, we would do something like `mat.decalMap = new BABYLON.DecalMapConfiguration(mat);` which is ok in javascript, but in Typescript it would fail because `decalMap` would not exist on the material.

I think the problem was the same for the detail map, which we finally added in the base standard and PBR material classes.

Reply:
Yup let s chat on Monday with @RaananW as well as it would be great to find a nice pattern for this.

The generator could inject the plugin for all materials when created the first time maybe ? and contain accessor by materials ?

Reply:
The generator is material independent, these are two unrelated things, as it could be used for something other than the decal map functionality (the generator is renamed `MeshUVSpaceRenderer` for now).

Reply:
Let's discuss this offline :-) I need to fully understand the usecase

Reply:
We found a solution :)